packages/flutter/lib/src/rendering/reactive_text_span.dart

```diff
@@ -0,0 +1,145 @@
+         semanticsLabel: semanticsLabel,
+       );
+
+  /// A gesture recognizer that will receive events that hit this span.
+  ///
+  /// [ReactiveTextSpan] itself does not implement hit testing or event dispatch.
+  /// The object that manages the [ReactiveTextSpan] painting is also responsible
+  /// for dispatching events. In the rendering library, that is the
+  /// [RenderParagraph] object, which corresponds to the [RichText] widget in the
+  /// widgets layer; these objects do not bubble events in [InlineSpan]s, so a
+  /// [recognizer] is only effective for events that directly hit the [text] of
+  /// that [ReactiveTextSpan], not any of its [children].
+  ///
+  /// [ReactiveTextSpan] also does not manage the lifetime of the gesture
+  /// recognizer. The code that owns the [GestureRecognizer] object must call
+  /// [GestureRecognizer.dispose] when the [ReactiveTextSpan] object is no longer
+  /// used.
+  ///
+  /// {@tool snippet}
+  ///
+  /// This example shows how to manage the lifetime of a gesture recognizer
+  /// provided to an [ReactiveTextSpan] object. It defines a `BuzzingText` widget
+  /// which uses the [HapticFeedback] class to vibrate the device when the user
+  /// long-presses the "find the" span, which is underlined in wavy green. The
+  /// hit-testing is handled by the [RichText] widget.
+  ///
+  /// ```dart
+  /// class BuzzingText extends StatefulWidget {
+  ///   @override
+  ///   _BuzzingTextState createState() => _BuzzingTextState();
+  /// }
+  ///
+  /// class _BuzzingTextState extends State<BuzzingText> {
+  ///   LongPressGestureRecognizer _longPressRecognizer;
+  ///
+  ///   @override
+  ///   void initState() {
+  ///     super.initState();
+  ///     _longPressRecognizer = LongPressGestureRecognizer()
+  ///       ..onLongPress = _handlePress;
+  ///   }
+  ///
+  ///   @override
+  ///   void dispose() {
+  ///     _longPressRecognizer.dispose();
+  ///     super.dispose();
+  ///   }
+  ///
+  ///   void _handlePress() {
+  ///     HapticFeedback.vibrate();
+  ///   }
+  ///
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return Text.rich(
+  ///       TextSpan(
+  ///         text: 'Can you ',
+  ///         style: TextStyle(color: Colors.black),
+  ///         children: <InlineSpan>[
+  ///           ReactiveTextSpan(
+  ///             text: 'find the',
+  ///             style: TextStyle(
+  ///               color: Colors.green,
+  ///               decoration: TextDecoration.underline,
+  ///               decorationStyle: TextDecorationStyle.wavy,
+  ///             ),
+  ///             recognizer: _longPressRecognizer,
+  ///           ),
+  ///           TextSpan(
+  ///             text: ' secret?',
+  ///           ),
+  ///         ],
+  ///       ),
+  ///     );
+  ///   }
+  /// }
+  /// ```
+  /// {@end-tool}
+  @override
+  // This field is overridden only to override the documentation.
+  GestureRecognizer? get recognizer => super.recognizer;
```

Comment:
If we are deprecating the gesture recognizer in the parent class `TextSpan`, we should also begin to move the canonical implementation to this class (if reasonable) to make full removal possible in the future. You don't necessarily have to do it in this PR, but maybe a TODO to make note that we should do it when it is less messy to do so.

Reply:
Agreed. And I would add that deprecating something before having a complete alternative doesn't help much.

Reply:
@mdebbar What do you mean by "before having a complete alternative"? This is intended to be a "rename": Use `ReactiveTextSpan.recognizer` instead of `TextSpan.recognizer`, and nothing else has to be changed.